syntax = "proto3";

import "google/protobuf/empty.proto";

package heartbeat;

service Heartbeat {
  rpc Heartbeat(HeartbeatRequest) returns (google.protobuf.Empty);
  rpc RequestVote(VoteRequest) returns (VoteResponse);
}

message HeartbeatRequest {
  string service_identifier = 1;
  string term = 2;
  string log_index = 3;
  string log_term = 4;
}

message HeartbeatResponse { string message = 1; }

message VoteRequest {
  int32 term = 1;
  string candidate_id = 2;
  int32 last_log_index = 3;
  int32 last_log_term = 4;
}

message VoteResponse {
  int32 term = 1;
  bool vote_granted = 2;
}